crates/bevy_math/src/curves/variable_linear.rs

```diff
@@ -0,0 +1,210 @@
+use super::Curve;
+use crate::interpolation::Lerp;
+
+// TODO: Curve/Clip need a validation during deserialization because they are
+// structured as SOA (struct of arrays), so the vec's length must match
+
+// TODO: impl Serialize, Deserialize
+/// Curve with sparse keyframes frames, in another words a curve with variable frame rate;
+///
+/// Very useful curve, because can accommodate the output of a linear reduction keyframe algorithm
+/// to lower the memory foot print. But as a down side require the use of keyframe cursor, and
+/// loses performance when the curve frame rate is higher than the curve sampling frame rate
```

Comment:
```suggestion
/// Curve with sparse keyframes, in another words a curve with variable frame rate;
///
/// This is a very useful curve, because it can accommodate the output of a linear reduction keyframe algorithm
/// to lower the memory foot print. As a down side it requires the use of a keyframe cursor, and
/// loses performance when the curve frame rate is higher than the curve sampling frame rate.
```